### PR TITLE
[RLlib] Try to checkpoint a durable policy name

### PR DESCRIPTION
**Pull request by gjoliver**
## Why are these changes needed?

instead of the raw policy class.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [*] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/registry.py [hunk 1]
```diff
@@ -282,3 +283,40 @@ def _get_algorithm_class(alg: str, return_config=False) -> type:
...
+
+def get_policy_class(name: str):
+    if name not in POLICIES:
+        return None
+
+    path = POLICIES[name]
+    module = importlib.import_module("ray.rllib.algorithms." + path)
+
+    if not hasattr(module, name):
+        return None
```
Comment (kouroshHakha) [inline rllib/algorithms/registry.py:320]: Do you wanna raise errors instead? is there a reason you are returning None?

File: rllib/policy/policy.py [hunk 2]
```diff
@@ -102,21 +102,42 @@ def __eq__(self, other: "PolicySpec"):
         )
 
     def serialize(self) -> Dict:
+        from ray.rllib.algorithms.registry import get_policy_class_name
```
Comment (kouroshHakha) [inline rllib/policy/policy.py:105]: Add the import to the top of the file

File: rllib/algorithms/registry.py [hunk 3]
```diff
@@ -282,3 +283,40 @@ def _get_algorithm_class(alg: str, return_config=False) -> type:
...
+    if policy_class.__name__ in POLICIES:
+        return policy_class.__name__
+    return None
+
+
+def get_policy_class(name: str):
+    if name not in POLICIES:
+        return None
+
+    path = POLICIES[name]
```
Comment (kouroshHakha) [inline rllib/algorithms/registry.py:316]: What if the name does not exist? e.g. custom policy?

**Review by kouroshHakha — APPROVED** [review]
left some comments.

File: rllib/algorithms/registry.py (hunk 3 above)
Comment (gjoliver) [reply rllib/algorithms/registry.py:316]: return None above.
in this case, we will checkpoint the raw policy class for the user and print a warning message.
I feel like if you want to use custom policy, then you'd better handle compatibility yourself?

File: rllib/policy/policy.py (hunk 2 above)
Comment (gjoliver) [reply rllib/policy/policy.py:105]: I can't 😭 
circular dependency somehow ...

**Pull request merged**